### PR TITLE
Add frontend ui codecov

### LINKED ISSUE #2251: Implement a minimum Code Coverage Report to meshery project
#### Current Behavior

We need to be able to see a Code Coverage Report and measure how many tests we have and those that are missing. This will allow us 

#### Desired Behavior

Have a Code Coverage Report on every PR and/or in the README.md.
We can start with the simplest implementation that measures the code coverage of unit tests or an only workflow. Then, we can create more issues to improve it.

---
[Optional]

#### Implementation

There are many tools that can be used: codecov or others. Feel free to choose the best.

### PR DESCRIPTION
**Pull request by shubhank-saxena**
**Description**

This PR fixes #2251 

**Notes for Reviewers**
@leecalcote kindly allow codecov as an application so that we can see the result of the report here

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@shubhank-saxena will you send a request for the app to be installed?

**Comment by leecalcote** [comment]
Ok, the CodeCov app is now installed at the `layer5io` org level. 👍

File: .codecov.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+coverage:
+  status:
+    patch: off
+    project:
+      default:
+        threshold: 2
+      ui:
+        target: 40%
```
Comment (leecalcote) [inline .codecov.yml:8]: We will be no where near these targets.

**Review by leecalcote — APPROVED** [review]

**Comment by leecalcote** [comment]
@ramrodo, I assume [this failed check](https://github.com/layer5io/meshery/pull/2254/checks?check_run_id=1710696918) isn't due to this PR.

**Comment by leecalcote** [comment]
There wasn’t a very specific or actionable error description in the failed check, so I’m re-running the CI job in case it was a glitch

**Comment by ramrodo** [comment]
@leecalcote @shubhank-saxena the error on CI is:

```
/home/runner/work/_temp/e796806c-0606-4729-a8bd-3a1cf0397754.sh: line 4: codecov: command not found
```

That is because we're trying to run **codecov** but it doesn't find the package in _node_modules_. So, as @Aisuko mentioned in [this comment](https://github.com/layer5io/meshery/issues/2251#issuecomment-760887131), we can add the Github Action for **codecov** and let it to execute the report.

I recommend removing that two **codecov** command and implement the GitHub Action in a later PR.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -139,10 +139,12 @@ jobs:
         cd ui
         npm i
         npm run ci-test-integration
+        codecov
```
Comment (ramrodo) [inline .github/workflows/ci.yml:142]: Remove this line and implement the **codecov** command with a GitHub Action in other PR.

```suggestion
        
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -139,10 +139,12 @@ jobs:
...
         cd ..
         cd provider-ui
         npm i
         npm test
+        codecov
```
Comment (ramrodo) [inline .github/workflows/ci.yml:147]: Remove this line and implement the **codecov** command with a GitHub Action in other PR.

```suggestion
        
```

**Review by ramrodo — CHANGES_REQUESTED** [review]

**Comment by shubhank-saxena** [comment]
> @leecalcote @shubhank-saxena the error on CI is:
> 
> ```
> /home/runner/work/_temp/e796806c-0606-4729-a8bd-3a1cf0397754.sh: line 4: codecov: command not found
> ```
> 
> That is because we're trying to run **codecov** but it doesn't find the package in _node_modules_. So, as @Aisuko mentioned in [this comment](https://github.com/layer5io/meshery/issues/2251#issuecomment-760887131), we can add the Github Action for **codecov** and let it to execute the report.
> 
> I recommend removing that two **codecov** command and implement the GitHub Action in a later PR.

@ramrodo the comment highlighted there is not actually a github action but github app. Also, if we even have another action for it, we still need to generate the coverage.xml report of every test.
Here is an example which shows how codecov is integrated with github actions (A different way) - https://github.com/codecov/codecov-action

**Comment by Aisuko** [comment]
Thanks, @shubhank-saxena  I have one repo can as an example, please kindly check the PR https://github.com/Aisuko/meshinfra/pull/25#issuecomment-759967937, and I saw `coverage1.xml` configuration can be optional. So, we should discuss this more and clearly how we can do this before we merge this one.

**Comment by shubhank-saxena** [comment]
> Thanks, @shubhank-saxena I have one repo can as an example, please kindly check the PR [Aisuko/meshinfra#25 (comment)](https://github.com/Aisuko/meshinfra/pull/25#issuecomment-759967937), and I saw `coverage1.xml` configuration can be optional. So, we should discuss this more and clearly how we can do this before we merge this one.

Ohh I see. Thank you for the link!

**Comment by shubhank-saxena** [comment]
> > @leecalcote @shubhank-saxena the error on CI is:
> > ```
> > /home/runner/work/_temp/e796806c-0606-4729-a8bd-3a1cf0397754.sh: line 4: codecov: command not found
> > ```
> > 
> > 
> > That is because we're trying to run **codecov** but it doesn't find the package in _node_modules_. So, as @Aisuko mentioned in [this comment](https://github.com/layer5io/meshery/issues/2251#issuecomment-760887131), we can add the Github Action for **codecov** and let it to execute the report.
> > I recommend removing that two **codecov** command and implement the GitHub Action in a later PR.
> 
> @ramrodo the comment highlighted there is not actually a github action but github app. Also, if we even have another action for it, we still need to generate the coverage.xml report of every test.
> Here is an example which shows how codecov is integrated with github actions (A different way) - https://github.com/codecov/codecov-action

This is redundant

**Comment by leecalcote** [comment]
Gents, it'd be good to get the approach designed and documented in here - https://docs.google.com/document/d/11nAxYtz2SUusCYZ0JeNRrOLIxkgmmbUVWz63MBZV2oE/edit#

**Review by ramrodo — APPROVED** [review]

**Comment by ramrodo** [comment]
> > @leecalcote @shubhank-saxena the error on CI is:
> > ```
> > /home/runner/work/_temp/e796806c-0606-4729-a8bd-3a1cf0397754.sh: line 4: codecov: command not found
> > ```
> > 
> > 
> > That is because we're trying to run **codecov** but it doesn't find the package in _node_modules_. So, as @Aisuko mentioned in [this comment](https://github.com/layer5io/meshery/issues/2251#issuecomment-760887131), we can add the Github Action for **codecov** and let it to execute the report.
> > I recommend removing that two **codecov** command and implement the GitHub Action in a later PR.
> 
> @ramrodo the comment highlighted there is not actually a github action but github app. Also, if we even have another action for it, we still need to generate the coverage.xml report of every test.
> Here is an example which shows how codecov is integrated with github actions (A different way) - https://github.com/codecov/codecov-action

@shubhank-saxena sounds good, Could you add that Github Action? Then we need to add the `CODECOV_TOKEN` to our Github Secrets (I already linked the Github App to the meshery project). @leecalcote I'll send you a slack message to add this Github Secret.

**Comment by leecalcote** [comment]
Org-wide `codecov_token` has been added.

**Pull request closed without merging**